docs/architecture/node-high-level.md

```diff
@@ -1 +1,117 @@
+
+The provider's identity agent then signs this proposal and the node invokes a service announcement to the Discovery API. The proposal becomes publicly available for everyone.
+
+## Communication channel
+
+Messaging channels enable nodes to exchange messages.
+
+A dialogue between nodes can be started over any type of messaging channel available to both nodes participating in a dialogue. One default channel type and associated communication protocol are be defined and supported on all Mysterium network clients.
+
+Each message sent over a channel is signed by the sender and then encrypted.
+
+Dialogues provide a structured way for two identities (peers) to exchange information needed to arrange payments for services provided and provision service sessions.
+
+### Session establishing
+
+To start a service session a client must request the creation of a session. The request includes a service proposal used to discover a service, a correlation ID used to correlate response, and an application specific data.
+
+After receiving this request, the provider may respond with an acceptance or refusal. In the case that a session opening was refused, a reason of refusal is also supplied. In the case that the opening of a channel is accepted, a session id is supplied and accompanied by application specific data supplied by the provider.
+
+A client should then use application-specific information received from a provider to try to establish a data carrying channel(s) and report back to the node. The client node then sends a message indicating confirmation of the successful opening of a session or failure to open a session to the provider.
+
+### Message broker
+
+At that moment Mysterium network uses centralized message broker for communicating between nodes in the network. It allows to send and receive messages from the registered network participants.
+
+## OpenVPN service
+
+In the current implementation VPN session between a service provider and service consumer established using OpenVPN solution.
+
+It creates a tunnel between provider and service, that is used for real data transfer.
+
+User IP-address is hidden for his requests through the OpenVPN tunnel. Destination services will only see IP-address of the service provider.
+
+All data transferred in this tunnel encrypted to prevent reading it by the unauthorized intermediate systems.
+
+## Blockchain
+
+Mysterium network highly depends on the blockchain technologies.
+
+In the Mysterium network, all network users have an account(identity) managed by the smart contract on Ethereum blockchain.
+
+Ethereum allows running decentralized code with smart contracts, enabling reliable services and payment handling.
+
+Identity service and database of registered identities ensures the proper identity
+acknowledgment between service provider and consumer.
+
+All processes in the Mysterium network, like payments and identity registration, done through the Ethereum blockchain smart contracts.
```

Comment:
All processes in the Mysterium network, like payments and identity registration are done through the Ethereum blockchain smart contracts.